diff --git a/ngx_http_fancyindex_module.c b/ngx_http_fancyindex_module.c
index 9675e92..0689b74 100644
--- a/ngx_http_fancyindex_module.c
+++ b/ngx_http_fancyindex_module.c
@@ -48,6 +48,7 @@ typedef struct {
     ngx_str_t  header;       /**< File name for header, or empty if none. */
     ngx_str_t  footer;       /**< File name for footer, or empty if none. */
     ngx_str_t  css_href;     /**< Link to a CSS stylesheet, or empty if none. */
+    ngx_flag_t show_dot_files;
 
     ngx_array_t *ignore;     /**< List of files to ignore in listings. */
 } ngx_http_fancyindex_loc_conf_t;
@@ -216,6 +217,13 @@ static ngx_command_t  ngx_http_fancyindex_commands[] = {
       offsetof(ngx_http_fancyindex_loc_conf_t, css_href),
       NULL },
 
+    { ngx_string("fancyindex_show_dot_files"),
+      NGX_HTTP_MAIN_CONF|NGX_HTTP_SRV_CONF|NGX_HTTP_LOC_CONF|NGX_CONF_FLAG,
+      ngx_conf_set_flag_slot,
+      NGX_HTTP_LOC_CONF_OFFSET,
+      offsetof(ngx_http_fancyindex_loc_conf_t, show_dot_files),
+      NULL },
+
     { ngx_string("fancyindex_ignore"),
       NGX_HTTP_MAIN_CONF|NGX_HTTP_SRV_CONF|NGX_HTTP_LOC_CONF|NGX_CONF_1MORE,
       ngx_http_fancyindex_ignore,
@@ -494,7 +502,10 @@ make_content_buf(
 
         len = ngx_de_namelen(&dir);
 
-        if (ngx_de_name(&dir)[0] == '.')
+        if (!alcf->show_dot_files && ngx_de_name(&dir)[0] == '.')
+            continue;
+
+        if (ngx_strcmp(ngx_de_name(&dir), ".") == 0 || ngx_strcmp(ngx_de_name(&dir), "..") == 0)
             continue;
 
 #if NGX_PCRE
@@ -1181,6 +1192,7 @@ ngx_http_fancyindex_create_loc_conf(ngx_conf_t *cf)
     conf->localtime    = NGX_CONF_UNSET;
     conf->name_length  = NGX_CONF_UNSET_UINT;
     conf->exact_size   = NGX_CONF_UNSET;
+    conf->show_dot_files = NGX_CONF_UNSET;
     conf->ignore       = NGX_CONF_UNSET_PTR;
 
     return conf;
@@ -1198,6 +1210,7 @@ ngx_http_fancyindex_merge_loc_conf(ngx_conf_t *cf, void *parent, void *child)
     ngx_conf_merge_value(conf->localtime, prev->localtime, 0);
     ngx_conf_merge_value(conf->exact_size, prev->exact_size, 1);
     ngx_conf_merge_uint_value(conf->name_length, prev->name_length, 50);
+    ngx_conf_merge_value(conf->show_dot_files, prev->show_dot_files, 0);
 
     ngx_conf_merge_str_value(conf->header, prev->header, "");
     ngx_conf_merge_str_value(conf->footer, prev->footer, "");
