diff --git a/ngx_http_fancyindex_module.c b/ngx_http_fancyindex_module.c
index 0d569fd..b9b43b8 100644
--- a/ngx_http_fancyindex_module.c
+++ b/ngx_http_fancyindex_module.c
@@ -161,6 +161,8 @@ typedef struct {
     ngx_str_t  css_href;     /**< Link to a CSS stylesheet, or empty if none. */
     ngx_str_t  time_format;  /**< Format used for file timestamps. */
 
+    ngx_flag_t show_dot_files;
+
     ngx_array_t *ignore;     /**< List of files to ignore in listings. */
 } ngx_http_fancyindex_loc_conf_t;
 
@@ -376,6 +378,13 @@ static ngx_command_t  ngx_http_fancyindex_commands[] = {
       offsetof(ngx_http_fancyindex_loc_conf_t, time_format),
       NULL },
 
+    { ngx_string("fancyindex_show_dot_files"),
+      NGX_HTTP_MAIN_CONF|NGX_HTTP_SRV_CONF|NGX_HTTP_LOC_CONF|NGX_CONF_FLAG,
+      ngx_conf_set_flag_slot,
+      NGX_HTTP_LOC_CONF_OFFSET,
+      offsetof(ngx_http_fancyindex_loc_conf_t, show_dot_files),
+      NULL },
+
     ngx_null_command
 };
 
@@ -654,7 +663,10 @@ make_content_buf(
 
         len = ngx_de_namelen(&dir);
 
-        if (ngx_de_name(&dir)[0] == '.')
+        if (!alcf->show_dot_files && ngx_de_name(&dir)[0] == '.')
+            continue;
+
+        if (ngx_strcmp(ngx_de_name(&dir), ".") == 0 || ngx_strcmp(ngx_de_name(&dir), "..") == 0)
             continue;
 
         if (alcf->hide_symlinks && ngx_de_is_link (&dir))
@@ -1332,6 +1344,7 @@ ngx_http_fancyindex_create_loc_conf(ngx_conf_t *cf)
     conf->hide_symlinks = NGX_CONF_UNSET;
     conf->show_path     = NGX_CONF_UNSET;
     conf->hide_parent   = NGX_CONF_UNSET;
+    conf->show_dot_files = NGX_CONF_UNSET;
 
     return conf;
 }
@@ -1362,6 +1375,8 @@ ngx_http_fancyindex_merge_loc_conf(ngx_conf_t *cf, void *parent, void *child)
     ngx_conf_merge_value(conf->hide_symlinks, prev->hide_symlinks, 0);
     ngx_conf_merge_value(conf->hide_parent, prev->hide_parent, 0);
 
+    ngx_conf_merge_value(conf->show_dot_files, prev->show_dot_files, 0);
+
     /* Just make sure we haven't disabled the show_path directive without providing a custom header */
     if (conf->show_path == 0 && conf->header.len == 0)
     {
